File: src/dataset/cpp/kindergarten-garden_exercism-chrisnp/claude/ClaudeSonnet4_kindergarten_garden_v4.cpp
#include "kindergarten_garden.h"

#include <stdexcept>

using std::array;
using std::string;

array<kindergarten_garden::Plants, 4> 
kindergarten_garden::plants(string const& garden, string const& student) {

    const size_t upRowIdx = 2 * (student[0] - 'A');
    const size_t dnRowIdx = garden.find('\n') + upRowIdx + 1;
    
    return {
        static_cast<Plants>(garden[upRowIdx]),
        static_cast<Plants>(garden[upRowIdx + 1]),
        static_cast<Plants>(garden[dnRowIdx]),
        static_cast<Plants>(garden[dnRowIdx + 1])
    };
}

File: src/dataset/Java/kindergarten-garden_exercism-uzilan/claude/ClaudeSonnet4_KindergartenGarden_v4.java
import java.util.ArrayList;
import java.util.List;

class KindergartenGarden {

    private final List<Plant[]> plants;

    KindergartenGarden(String garden) {
        String[] rows = garden.split("\n");
        plants = new ArrayList<>(rows.length);
        for (String row : rows) {
            plants.add(parseRow(row));
        }
    }

    private Plant[] parseRow(String row) {
        int length = row.length();
        Plant[] rowPlants = new Plant[length];
        for (int i = 0; i < length; i++) {
            rowPlants[i] = Plant.getPlant(row.charAt(i));
        }
        return rowPlants;
    }

    List<Plant> getPlantsOfStudent(String student) {
        final int studentIndex = student.charAt(0) - 'A';
        final int startIndex = studentIndex * 2;
        final int endIndex = startIndex + 2;
        
        List<Plant> result = new ArrayList<>(plants.size() * 2);
        for (Plant[] row : plants) {
            result.add(row[startIndex]);
            result.add(row[endIndex]);
        }
        return result;
    }
}